src/menu/MenuItem.ts

```diff
@@ -0,0 +1,186 @@
+import { assign } from '@dojo/core/lang';
+import { v } from '@dojo/widget-core/d';
+import { VirtualDomProperties } from '@dojo/widget-core/interfaces';
+import ThemeableMixin, { theme, ThemeableProperties } from '@dojo/widget-core/mixins/Themeable';
+import WidgetBase from '@dojo/widget-core/WidgetBase';
+import * as css from './styles/menu.m.css';
+import { Keys } from '../common/util';
+
+export type MenuItemType = 'item' | 'checkbox' | 'radio';
+
+/**
+ * @type MenuItemProperties
+ *
+ * Properties that can be set on a MenuItem component.
+ *
+ * @property active         Determines whether the item should receive focus.
+ * @property controls       The ID of an element that this input controls.
+ * @property disabled       Indicates whether the item is disabled.
+ * @property expanded       Indicates whether a widget controlled by `this` is expanded.
+ * @property focusable      Determines whether the item can receive focus with tab key.
+ * @property hasMenu        Indicates whether the widget is used as the label for a menu.
+ * @property hasPopup       Indicates whether the widget has a drop down child.
+ * @property id             Specifies the ID for the widget.
+ * @property index          Specifies the index of the item within a parent menu.
+ * @property onClick        Called when the widget is activated via a click or space key.
+ * @property onKeyDown      Called when keys are pressed while the widget has focus.
+ * @property onMouseDown    Called during a mousedown event.
+ * @property properties     Additional properties for the widget's vnode.
+ * @property selected       Determines whether the widget is marked as selected.
+ * @property tag            The HTML tag name to use for the widget's vnode. Defaults to 'a'.
+ * @property type           The type of the menu item. Defaults to 'item'.
```

Comment:
It was `role` originally, but @smhigley and I agreed that allowing users to specify `item`, `checkbox`, or `radio` instead of `menuitem`, `menuitemcheckbox` or `menuitemradio` would be better, in which case `role` became somewhat misleading.